src/hash.c

```diff
@@ -15,3 +15,17 @@ unsigned long djb2(unsigned char *str) {
     // NOLINTEND
 }
 
+// A version that takes in a generic array
+unsigned long djb2_bytes(void* bytes, size_t num) {
+    // NOLINTBEGIN
+    size_t hash = 5381;
```

Comment:
technically it would probably make more sense for this to be `unsigned long` but for practical purposes it's probably fine 

in gcc unsigned long is typically 32 bit and size_t is typically 64 bit